### PR TITLE
feat(localization): apply preferred locale on documents

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

For legal reason, some companies had to generate invoices in their own languages. Unfortunately Lago is supporting only one language that do not fit their need.

## Description

Before generating PDF documents document locale is set based on customer/organization configuration. Also, this PR covers translations of all .slim files that are stored in Lago

### CODE REVIEW

File: spec/models/customer_spec.rb [hunk 1]
```diff
@@ -85,6 +85,30 @@
     end
   end
 
+  describe 'preferred_document_locale' do
+    subject(:customer) do
+      described_class.new(
+        organization: organization,
```
Comment (jdenquin) [inline spec/models/customer_spec.rb:91]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/credit_notes/generate_service_spec.rb [hunk 2]
```diff
@@ -38,6 +38,15 @@
       expect(result.credit_note.file).to be_present
     end
 
+    context 'with preferred locale' do
+      before { customer.update!(document_locale: 'fr') }
+
+      it 'sets the correct document locale' do
+        expect { credit_note_generate_service.call }
+          .to change { I18n.locale }.from(:en).to(:fr)
```
Comment (jdenquin) [inline spec/services/credit_notes/generate_service_spec.rb:46]: RSpec/ExpectChange: Prefer `change(I18n, :locale)`. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ExpectChange)

File: spec/services/invoices/generate_service_spec.rb [hunk 3]
```diff
@@ -99,6 +99,15 @@
 
         expect(result.invoice.file).to be_present
       end
+
+      context 'with preferred locale' do
+        before { customer.update!(document_locale: 'fr') }
+
+        it 'sets the correct document locale' do
+          expect { invoice_generate_service.generate(invoice_id: invoice.id) }
+            .to change { I18n.locale }.from(:en).to(:fr)
```
Comment (jdenquin) [inline spec/services/invoices/generate_service_spec.rb:108]: RSpec/ExpectChange: Prefer `change(I18n, :locale)`. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ExpectChange)

File: app/models/customer.rb [hunk 4]
```diff
@@ -80,6 +80,12 @@ def editable?
       wallets.none?
   end
 
+  def preferred_document_locale
+    return document_locale&.to_sym if document_locale.present?
```
Comment (rsempe) [inline app/models/customer.rb:84]:
```suggestion
    return document_locale.to_sym if document_locale?
```

File: app/models/customer.rb [hunk 5]
```diff
@@ -80,6 +80,12 @@ def editable?
...
+
+    organization.document_locale&.to_sym
```
Comment (rsempe) [inline app/models/customer.rb:86]:
```suggestion
    organization.document_locale.to_sym
```

as organization.document_locale cannot be null?

File: app/services/invoices/generate_service.rb [hunk 6]
```diff
@@ -22,6 +22,8 @@ def generate(invoice_id:)
     private
 
     def generate_pdf(invoice)
+      I18n.locale = invoice&.customer&.preferred_document_locale || I18n.default_locale
```
Comment (rsempe) [inline app/services/invoices/generate_service.rb:25]: same here

File: app/models/customer.rb (hunk 5 above)
Comment (lovrocolic) [reply app/models/customer.rb]: fixed

File: app/models/customer.rb (hunk 4 above)
Comment (lovrocolic) [reply app/models/customer.rb]: refactored

File: app/services/credit_notes/generate_service.rb [hunk 7]
```diff
@@ -26,6 +26,8 @@ def call
     attr_reader :credit_note, :context
 
     def generate_pdf(credit_note)
+      I18n.locale = credit_note&.customer&.preferred_document_locale
```
Comment (rsempe) [inline app/services/credit_notes/generate_service.rb:29]: `credit_note` is always present here and the customer is always associated, no?

```suggestion
      I18n.locale = credit_note.customer.preferred_document_locale
```

File: app/services/invoices/generate_service.rb [hunk 8]
```diff
@@ -22,6 +22,8 @@ def generate(invoice_id:)
     private
 
     def generate_pdf(invoice)
+      I18n.locale = invoice&.customer&.preferred_document_locale
```
Comment (rsempe) [inline app/services/invoices/generate_service.rb:25]:
```suggestion
      I18n.locale = invoice.customer.preferred_document_locale
```

File: app/views/templates/credit_note.slim [hunk 9]
```diff
@@ -326,61 +326,61 @@ html
           .body-2 = customer.email
 
       .mb-24
-        h2.title-2.mb-8 = total_amount.format
+        h2.title-2.mb-8 = total_amount.format(format: I18n.t('money.format'), decimal_mark: I18n.t('money.decimal_mark'), thousands_separator: I18n.t('money.thousands_separator'))
         .body-1
           - if credited? && refunded?
-            | Credited on customer balance and refunded on #{issuing_date.strftime('%b %d, %Y')}
+            = I18n.t('credit_note.credited_refunded_notice', issuing_date: I18n.l(issuing_date, format: :default))
```
Comment (rsempe) [inline app/views/templates/credit_note.slim:332]: Is it mandatory to put `format: :default` or used by default?

File: config/locales/fr.yml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+      - Mardi
+      - Mercredi
+      - Jeudi
+      - Vendredi
+      - Samedi
+    formats:
+      default: '%d %b %Y'
+    month_names:
+      -
+      - Javien
```
Comment (rsempe) [inline config/locales/fr.yml:41]:
```suggestion
      - Janvier
```

File: config/locales/fr.yml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+      - Lundi
+      - Mardi
+      - Mercredi
+      - Jeudi
+      - Vendredi
+      - Samedi
+    formats:
+      default: '%d %b %Y'
+    month_names:
+      -
```
Comment (rsempe) [inline config/locales/fr.yml:40]: Why this line? 🤔

File: app/services/credit_notes/generate_service.rb (hunk 7 above)
Comment (lovrocolic) [reply app/services/credit_notes/generate_service.rb]: fixed

File: app/services/invoices/generate_service.rb (hunk 8 above)
Comment (lovrocolic) [reply app/services/invoices/generate_service.rb]: fixed

File: config/locales/fr.yml (hunk 10 above)
Comment (lovrocolic) [reply config/locales/fr.yml]: fixed

File: config/locales/fr.yml (hunk 11 above)
Comment (lovrocolic) [reply config/locales/fr.yml:40]: Not sure tbh, but has to be there. E.g. https://github.com/svenfuchs/rails-i18n/blob/master/rails/locale/en.yml

File: app/views/templates/credit_note.slim (hunk 9 above)
Comment (lovrocolic) [reply app/views/templates/credit_note.slim:332]: yep in this case, since default is just key name in yml file

File: config/locales/fr.yml (hunk 11 above)
Comment (rsempe) [reply config/locales/fr.yml:40]: fair enough 😄

**Review by rsempe — APPROVED** [review]

**Pull request merged**